### PR TITLE
feat: update service for fetch data

### PR DESCRIPTION
**Pull request by mthanh2209**

### CODE REVIEW

File: javascript-practice/src/javascripts/services/service.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+	/**
+	 * Method to send an HTTP request to the API endpoint.
+	 * @param {String} path - The endpoint path for the request.
+	 * @param {String} method - The HTTP method (GET, POST, PUT, DELETE, etc.).
+	 * @param {Object} body - The request body (optional).
+	 * @return {Promise} A promise that resolves to the server response data.
+	 * @throws {Error} If the request was not successful.
+	 */
+	sendRequest = async (id, method, body) => {
+		const url = `${this.apiUrl}${this.path}${id ? `/${id}` : ""}`;
```
Comment (hungnguyenhuu-agilityio) [inline javascript-practice/src/javascripts/services/service.js:24]: This one looks complex. Please add it to the helpers or s.thing like that.

File: javascript-practice/src/javascripts/services/service.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+	getList = () => this.sendRequest(null, "get");
+
+	/**
+	 * Builds a URL for a specific resource ID.
+	 * @param {string|null} id - The ID of the resource (optional).
+	 * @returns {string} The constructed URL.
+	 */
+	buildUrl(id) {
+		return `${this.apiUrl}${this.path}${id ? `/${id}` : ""}`;
+	}
```
Comment (hungnguyenhuu-agilityio) [inline javascript-practice/src/javascripts/services/service.js:22]: This one should be put on helpers or utilities

File: javascript-practice/src/javascripts/utilities/url.js [hunk 3]
```diff
@@ -0,0 +1,14 @@
+class Utilities {
```
Comment (hungnguyenhuu-agilityio) [inline javascript-practice/src/javascripts/utilities/url.js:1]: Should not define the utilities as a class

Comment (hungnguyenhuu-agilityio) [reply javascript-practice/src/javascripts/utilities/url.js:1]: https://www.geeksforgeeks.org/functional-programming-pure-and-impure-functions/

Comment (mthanh2209) [reply javascript-practice/src/javascripts/utilities/url.js:1]: I fixed it

File: javascript-practice/src/javascripts/utilities/url.js [hunk 4]
```diff
@@ -0,0 +1,11 @@
+/**
+ * Build a URL by combining the API URL, path, and optional ID.
+ *
+ * @param {string} apiUrl - The base API_URL.
+ * @param {string} path - The path to append to the base URL.
+ * @param {string} [id] - (Optional) An ID to append to the path (if provided).
+ * @returns {string} The constructed URL.
+ */
+export function BuildUrl(apiUrl, path, id) {
```
[the remark this answers is not in the document]
Comment (mthanh2209) [reply javascript-practice/src/javascripts/utilities/url.js]: Sorry, I fixed it

**Pull request merged**